linera-execution/src/wasm/wasmer.rs

```diff
@@ -273,27 +267,65 @@ impl From<wasmer::RuntimeError> for ExecutionError {
 }
 
 /// Serialized bytes of a compiled contract bytecode.
-///
-/// Each [`Module`] needs to be compiled with a separate [`Engine`] instance, otherwise Wasmer
-/// [panics](https://docs.rs/wasmer-middlewares/3.3.0/wasmer_middlewares/metering/struct.Metering.html#panic)
-/// because fuel metering is configured and used across different modules.
 pub struct CachedContractModule {
     compiled_bytecode: Bytes,
 }
 
+pub fn add_metering(bytecode: Bytecode) -> anyhow::Result<Bytecode> {
```

Comment:
Nit: adding documentation

```suggestion
/// Changes an input `bytecode` by injecting instrumentation to track fuel consumption.
///
/// Returns a new [`Bytecode`] with an extra global variable exported with the value of the
/// available fuel, which is updated during execution.
pub fn add_metering(bytecode: Bytecode) -> anyhow::Result<Bytecode> {
```